### PR TITLE
[Docs] Update udev rules to use TAG+="uaccess".

### PR DESCRIPTION
**Pull request by petejohanson**
## Description

* udev for several years now has supported "dynamic" permissions
  for device access for the logged in user, that avoids needing
  custom groups, or making devices world writable.
* Update the docs for udev rules to use the "newer" `uaccess`
  approach.

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [ ] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [x] Documentation

## Issues Fixed or Closed by This PR

* N/A

## Checklist

- [ ] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

**Comment by petejohanson** [comment]
> `lib/python/qmk/cli/doctor.py` needs updating to reflect these docs changes, though this might then cause issue as how do we tell users they need to update existing files content.

@zvecr Any objections to updating that Python code to be able to fallback to detecting the "old" version, and allowing it, but warning in that scenario about updating them?

File: docs/faq_build.md [hunk 1]
```diff
@@ -28,23 +28,23 @@ sudo udevadm trigger
...
-SUBSYSTEMS=="usb", ATTRS{idVendor}=="03eb", ATTRS{idProduct}=="2ffb", MODE:="0666"
+SUBSYSTEMS=="usb", ATTRS{idVendor}=="03eb", ATTRS{idProduct}=="2ffb", TAG+="uaccess", RUN{builtin}+="uaccess"
 # Atmel ATMega32U2
-SUBSYSTEMS=="usb", ATTRS{idVendor}=="03eb", ATTRS{idProduct}=="2ff0", MODE:="0666"
+SUBSYSTEMS=="usb", ATTRS{idVendor}=="03eb", ATTRS{idProduct}=="2ff0", TAG+="uaccess", RUN{builtin}+="uaccess"
 ```
 
 **/etc/udev/rules.d/52-tmk-keyboard.rules:**
 ```
 # tmk keyboard products     https://github.com/tmk/tmk_keyboard
```
Comment (Erovia) [inline docs/faq_build.md:40]:
```suggestion

```

File: docs/faq_build.md [hunk 2]
```diff
@@ -28,23 +28,23 @@ sudo udevadm trigger
...
-SUBSYSTEMS=="usb", ATTRS{idVendor}=="feed", MODE:="0666"
+SUBSYSTEMS=="usb", ATTRS{idVendor}=="feed", TAG+="uaccess", RUN{builtin}+="uaccess"
```
Comment (Erovia) [inline docs/faq_build.md:41]:
```suggestion

```
This rule is totally useless.

File: docs/faq_build.md [hunk 3]
```diff
@@ -28,23 +28,23 @@ sudo udevadm trigger
...
 ```
 **/etc/udev/rules.d/54-input-club-keyboard.rules:**
 
 ```
 # Input Club keyboard bootloader
-SUBSYSTEMS=="usb", ATTRS{idVendor}=="1c11", MODE:="0666"
+SUBSYSTEMS=="usb", ATTRS{idVendor}=="1c11", TAG+="uaccess", RUN{builtin}+="uaccess"
```
[the remark this answers is not in the document]
Comment (petejohanson) [reply docs/faq_build.md]: Done.

File: docs/faq_build.md [hunk 4]
```diff
@@ -28,23 +28,23 @@ sudo udevadm trigger
...
 ```
 
 **/etc/udev/rules.d/55-caterina.rules:**
```
[the remark this answers is not in the document]
Comment (petejohanson) [reply docs/faq_build.md:45]: Done.

File: docs/faq_build.md (hunk 2 above)
Comment (petejohanson) [reply docs/faq_build.md]: Removed.

File: docs/faq_build.md (hunk 1 above)
Comment (petejohanson) [reply docs/faq_build.md]: Removed.

**Pull request merged**